salt/proxy/restconf.py

```diff
@@ -0,0 +1,222 @@
+"""
+Proxy Minion to manage Restconf Devices
+
+:codeauthor: Jamie (Bear) Murphy <jamiemurphyit@gmail.com>
+:maturity:   new
+:platform:   any
+
+Usage
+=====
+
+.. note::
+
+    To be able to use this module you need to enable to RESTCONF on your device
+    and having https enabled.
+
+    Cisco Configuration example:
+
+    .. code-block:: bash
+
+        switch# conf t
+        switch(config)# restconf
+        switch(config)# ip http secure-server
+
+.. note::
+
+    RESTCONF requires modern OS distributions.
+    This plugin has been written specifically to use JSON Restconf endpoints
+
+Pillar
+------
+
+The ``restconf`` proxy configuration requires the following parameters in order
+to connect to the network switch:
+
+transport: ``https``
+    Specifies the type of connection transport to use. Valid values for the
+    connection are ``http``, and  ``https``.
+
+host: ``localhost``
+    The IP address or DNS host name of the connection device.
+
+username: ``admin``
+    The username to pass to the device to authenticate the RESTCONF requests.
+
+password:
+    The password to pass to the device to authenticate the RESTCONF requests.
```

Comment:
@barbaricyawps / @ScriptAutomate Do you have a suggestion for how to improve this wording? "The password to pass to the device" and "The username to pass to the device" feels weird to me.

Reply:
refactored